packages/compat/plugin-swc/src/utils.ts

```diff
@@ -184,6 +186,34 @@ const getCoreJsVersion = (corejsPkgPath: string) => {
   }
 };
 
+const isPrimitiveTransformImport = (
+  items: unknown[],
+): items is Array<TransformImport> =>
+  items.every(
+    (item) => Object.prototype.toString.call(item) === '[object Object]',
+  );
+
+const applyTransformImportChain = (
```

Comment:
- No need to do recursive calls after correct the types
- No need to pass `default` value, always defaults to `[]`

```suggestion
const reduceTransformImportConfig = (
  options: NormalizedSourceConfig['transformImport'],
): TransformImport[] | false => {
  if (options === false || !options) {
    return false;
  }

  let imports: TransformImport[] = [];
  for (const item of castArray(options)) {
    if (isFunction(item)) {
      imports = item(imports) ?? imports;
    } else {
      imports.push(item);
    }
  }
  return imports;
};
```